bin/oxstore

```diff
@@ -0,0 +1,304 @@
+        else:
+            lf.s3_key(bucket).get_file(fp=sys.stdout)
+
+
+def do_cat_local(log_path, type_name, start_dt, end_dt):
+    "Run the cat action against local log path"
+    log_files = find_log_files_in_path(log_path, type_name, start_dt, end_dt)
+
+    log_files.sort(key=lambda f: f.sort_dt)
+
+    for lf in log_files:
+        if lf.bzip:
+            decompressor = bz2.BZ2Decompressor()
+        else:
+            decompressor = None
+
+        with io.open(lf.get_local_file_path(log_path), "rb") as f:
+            if decompressor:
+                sys.stdout.write(decompressor.decompress(f.read()))
+            else:
+                sys.stdout.write(f.read())
+
+
+def do_zip(log_path):
+    log.debug("Listing log files for %r", log_path)
+    log_files = store.list_log_files(log_path)
+
+    for log_file in store.filter_log_files_for_zipping(log_files):
+        log.info("Zipping %s", log_file.file_path)
+
+        store.zip_log_file(log_file, log_path)
+
+
+def do_prune(log_path, retain_days):
+    min_archive_date = (datetime.datetime.utcnow() - datetime.timedelta(days=retain_days)).date()
+
+    for dirpath, dirnames, filenames in os.walk(log_path):
+        try:
+            dir_date = datetime.datetime.strptime(os.path.basename(dirpath), '%Y%m%d').date()
+        except ValueError:
+            continue
+
+        if dir_date < min_archive_date:
+            for filename in filenames:
+                full_path = os.path.join(dirpath, filename)
+                log.info("Removing %s", full_path)
+                os.unlink(full_path)
+
+            os.rmdir(dirpath)
+
+
+def parse_date_range_arguments(parser, args):
+    if args.start is None:
+        start_date = datetime.datetime.utcnow().replace(
+            hour=0, minute=0, second=0, microsecond=0)
+    else:
+        try:
+            start_date = store.parse_date_range_argument(args.start)
+        except store.InvalidDateError:
+            parser.error("Bad format for start date. Try YYYYMMDD")
+
+    if args.end is None:
+        end_date = start_date + datetime.timedelta(hours=23, minutes=59, seconds=59)
+    else:
+        try:
+            end_date = store.parse_date_range_argument(args.end)
+        except store.InvalidDateError:
+            parser.error("Bad format for end date. Try YYYYMMDD")
+
+    return start_date, end_date
+
+
+def open_bucket(bucket_name):
+    region_name = os.environ.get('AWS_DEFAULT_REGION', 'us-east-1')
+
+    # We need to specify calling_format to get around bugs in having a '.' in
+    # the bucket name
+    conn = boto.s3.connect_to_region(region_name, calling_format=OrdinaryCallingFormat())
+
+    bucket = conn.get_bucket(bucket_name)
+    if not bucket:
+        parser.error("Missing bucket {}".format(bucket_name))
+
+    return bucket
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Manage blueox logs')
+    parser.add_argument('--verbose', '-v', action='count')
+
+    subparsers = parser.add_subparsers(
```

Comment:
It would be nicer to use and easier to read the code if these options had help strings